unic-locale-impl/src/extensions/transform.rs

```diff
@@ -59,11 +56,15 @@ impl TransformExtensionList {
         Ok(())
     }
 
-    pub fn set_tfield(&mut self, tkey: &str, tvalue: Vec<&str>) -> Result<(), LocaleError> {
-        let tkey = parse_tkey(tkey)?;
+    pub fn set_tfield<S: AsRef<[u8]>>(
+        &mut self,
+        tkey: S,
+        tvalue: Vec<S>,
```

Comment:
Do you really need a vec? Would this suffice with a `&[S]`? seems like you throw the vec away anyways.